helm-charts/splunk-otel-collector/values.yaml

```diff
@@ -350,13 +350,19 @@ agent:
       memory: 500Mi
 
   # To collect container logs and journald logs, it will run the agent as a root user.
-  # To run it as non root user, uncomment below `securityContext` options.
+  # To run it as non root user, change runAsUser and runAsGroup to non-zero value (e.g. 20000) and change runAsNonRoot to true.
   # Setting runAsUser and runAsGroup to a non root user enables an init container that patches group
   # permissions of container logs directories on the host filesystem to make logs readable by this non root user.
+  # NOTE: during internal tests running this container as a non-root user with cri-o runntime didn't work.
```

Comment:
How impactful is this? Do many users run this with cri-o runtime? If this is impactful is there an issue filed to follow up and resolve?

Reply:
Yes, Openshift by default uses CRI-O runtime. So, it is widely used.

Reply:
So for now, most users shouldn't use the functionality of this PR, right? The belief is also that the problem is in CRI-O, and nothing to do with our side, correct? Just trying to make sure I understand properly.

If these things are the case it should be made pretty clear to customers. We don't want anyone trying to actually use this if we know it's broken.

Reply:
> So for now, most users shouldn't use the functionality of this PR, right? 

Yes, we can make it more clear who should not enable this feature

>  The belief is also that the problem is in CRI-O, and nothing to do with our side.

Correct. I can brief you about what is expected and what is going wrong.

ACL(Access control lists) can add read, write or execute permission to specified users or groups.
When the patch-log-directory container is executed, it applied default ACLs to /var/log directory. So, when any new file/directory is created inside /var/log directory, it will inherit those ACLs. It is working as expected in docker/containerd runtime. But, not working in cri-o runtime.

```bash
bash-4.4# getfacl /var/log/pods/default_sck-splunk-otel-collector-agent-nspsk_ca2e984d-8338-4561-8337-d371d832f6af/otel-collector
getfacl: Removing leading '/' from absolute path names
# file: var/log/pods/default_sck-splunk-otel-collector-agent-nspsk_ca2e984d-8338-4561-8337-d371d832f6af/otel-collector
# owner: root
# group: root
user::rwx
group::r-x
group:20000:r-x
mask::r-x
other::r-x
default:user::rwx
default:group::r-x
default:group:20000:r-x
default:mask::r-x
default:other::r-x

bash-4.4# getfacl /var/log/pods/default_sck-splunk-otel-collector-agent-nspsk_ca2e984d-8338-4561-8337-d371d832f6af/otel-collector/0.log 
getfacl: Removing leading '/' from absolute path names
# file: var/log/pods/default_sck-splunk-otel-collector-agent-nspsk_ca2e984d-8338-4561-8337-d371d832f6af/otel-collector/0.log
# owner: root
# group: root
user::rw-
group::r-x                      #effective:---
group:20000:r-x                 #effective:---
mask::---
other::---
```

If I manually create a file there, it inherits corrects ACLs

```sh
bash-4.4# touch sample.txt
bash-4.4# getfacl /var/log/pods/default_sck-splunk-otel-collector-agent-nspsk_ca2e984d-8338-4561-8337-d371d832f6af/otel-collector/sample.txt 
getfacl: Removing leading '/' from absolute path names
# file: var/log/pods/default_sck-splunk-otel-collector-agent-nspsk_ca2e984d-8338-4561-8337-d371d832f6af/otel-collector/sample.txt
# owner: root
# group: root
user::rw-
group::r-x                      #effective:r--
group:20000:r-x                 #effective:r--
mask::r--
other::r--
```

Reply:
That makes sense! Thanks for filing an issue against CRI-O as well!

I'll include the issue here as well just for future reference: https://github.com/cri-o/cri-o/issues/6519